### ISSUE TITLE
prefer-array-literal: catch new Array(10) but not Array(10)

### ISSUE THREAD

**Issue by sangheestyle** [Microsoft/tslint-microsoft-contrib#839]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.1.0
-   **TSLint version**: 5.11.0
-   **TypeScript version**: 3.3.3333
-   **Running TSLint via**: (pick one) Node.js

#### TypeScript code being linted

```tsx
const caseA = new Array(10);
const caseB = Array(10);
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint-microsoft-contrib/latest"]
}
```

#### Actual behavior

With `tslint -c tslint.json -p tsconfig.json`, `Array(10)` didn't detected.

```
ERROR: ... : Replace Array constructor with an array literal: new Array(10)
```

#### Expected behavior

With `tslint -c tslint.json -p tsconfig.json`, both cases should be detected.

```
ERROR: ... : Replace Array constructor with an array literal: new Array(10)
ERROR: ... : Replace Array function with an array literal: Array(10)
```

**Comment by JoshuaKGoldberg** [comment]
Good find! Looks like the rule checks nodes that are `NewExpression`s but not `CallExpression`s.
